source/distributed/tria.cc

```diff
@@ -1689,34 +2225,48 @@ namespace parallel
 
       if (this->my_subdomain == 0)
         {
-          const unsigned int buffer_size_per_cell =
-            (cell_attached_data.cumulative_fixed_data_size > 0 ?
-               cell_attached_data.cumulative_fixed_data_size +
-                 sizeof(CellStatus) :
-               0);
-
           std::string   fname = std::string(filename) + ".info";
           std::ofstream f(fname.c_str());
-          f << "version nproc attached_bytes n_attached_objs n_coarse_cells"
-            << std::endl
-            << 2 << " "
+          f << "version nproc n_attached_objs n_coarse_cells" << std::endl
+            << 3 << " "
             << Utilities::MPI::n_mpi_processes(this->mpi_communicator) << " "
-            << buffer_size_per_cell << " "
             << cell_attached_data.pack_callbacks.size() << " "
             << this->n_cells(0) << std::endl;
         }
 
-      if (cell_attached_data.cumulative_fixed_data_size > 0)
+#  ifdef DEBUG
+      // each cell should have been flagged `CELL_PERSIST`
+      for (auto &qcrel : local_quadrant_cell_relations)
         {
-          const_cast<
-            dealii::parallel::distributed::Triangulation<dim, spacedim> *>(this)
-            ->attach_mesh_data();
+          const auto &cell_status = std::get<1>(qcrel);
+          Assert(
+            (cell_status ==
```

Comment:
if you inline the call to `get<1>` into this place, then the body of the loop is really just an assert. the compiler will optimize this away in release mode, and then you can also remove the `#ifdef ... #endif` around this.